test/arbitrage/BancorArbitrage.ts

```diff
@@ -0,0 +1,358 @@
+            await expect(
+                Contracts.TestBancorArbitrage.deploy(
+                    bnt.address,
+                    bancorV2.address,
+                    bancorV3.address,
+                    uniswapV2Router.address,
+                    ZERO_ADDRESS,
+                    sushiswapV2Router.address
+                )
+            ).to.be.revertedWithError('InvalidAddress');
+        });
+
+        it('should revert when initializing with an invalid Sushiswap router contract', async () => {
+            await expect(
+                Contracts.TestBancorArbitrage.deploy(
+                    bnt.address,
+                    bancorV2.address,
+                    bancorV3.address,
+                    uniswapV2Router.address,
+                    uniswapV3Router.address,
+                    ZERO_ADDRESS
+                )
+            ).to.be.revertedWithError('InvalidAddress');
+        });
+
+        it('should be initialized', async () => {
+            expect(await bancorArbitrage.version()).to.equal(1);
+        });
+
+        it('should revert when attempting to reinitialize', async () => {
+            await expect(bancorArbitrage.initialize()).to.be.revertedWithError(
+                'Initializable: contract is already initialized'
+            );
+        });
+    });
+
+    describe('rewards', () => {
+        it('should revert when a non-admin attempts to set the arbitrage rewards settings', async () => {
+            await expect(
+                bancorArbitrage.connect(nonOwner).setRewards(ArbitrageRewardsDefaults)
+            ).to.be.revertedWithError('AccessDenied');
+        });
+
+        it('should ignore setting to the same arbitrage rewards settings', async () => {
+            await bancorArbitrage.setRewards(ArbitrageRewardsDefaults);
+
+            const res = await bancorArbitrage.setRewards(ArbitrageRewardsDefaults);
+            await expect(res).not.to.emit(bancorArbitrage, 'RewardsUpdated');
+        });
+
+        it('should be able to set and update the arbitrage rewards settings', async () => {
+            await bancorArbitrage.setRewards(ArbitrageRewardsDefaults);
+
+            const res = await bancorArbitrage.rewards();
+            expect(res.percentagePPM).to.equal(30_000);
+
+            const resChanged = await bancorArbitrage.setRewards(ArbitrageRewardsChanged);
+            await expect(resChanged).to.emit(bancorArbitrage, 'RewardsUpdated');
+
+            const resUpdated = await bancorArbitrage.rewards();
+            expect(resUpdated.percentagePPM).to.equal(40_000);
+        });
+    });
+
+    describe('arbitrage', () => {
+        beforeEach(async () => {
+            await transfer(deployer, bnt, masterVault.address, AMOUNT.mul(10_000));
+            await bancorArbitrage.setRewards(ArbitrageRewardsDefaults);
+        });
+
+        // get all exchange ids (omit their names)
+        const exchangeIds = Object.values(ExchangeId).filter((key) => !isNaN(Number(key)));
```

Comment:
Very minor: `Number` works for floats as well, probably better to use `parseInt`.

Reply:
done